### ISSUE TITLE
Increased CoreDNS latency for TXT records.

### ISSUE THREAD

**Issue by michalgcc** [Azure/AKS#2358]
**What happened**:

After updating the AKS cluster Kubernetes version from 1.19.7 to 1.20.5 we noticed increased CoreDNS latency for TXT records.
On 1.20.5 the average latency is 5-20 ms. On 1.19.7 all responses were below 1 ms.

We are using the default AKS configuration:

```
.:53 {
    errors
    ready
    health
    kubernetes cluster.local in-addr.arpa ip6.arpa {
      pods insecure
      fallthrough in-addr.arpa ip6.arpa
    }
    prometheus :9153
    forward . /etc/resolv.conf
    cache 30
    loop
    reload
    loadbalance
    import custom/*.override
}
import custom/*.server
```

It looks like Kubernetes 1.20+ use CoreDNS 1.8.3 and Kubernetes 1.19.* use CoreDNS 1.6.6.

Could you let us know what might be the issue?

**What you expected to happen**:

Reduce CoreDNS latency for TXT records in Kubernetes 1.20.5.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.20.5
- Size of cluster (how many worker nodes are in the cluster?): 7
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): HTTP microservices.

**Comment by sdktr** [comment]
Any specific TXT record you noticed this with? Like some in-cluster service discovery record? With which record(s) can we reproduce?

**Comment by michalgcc** [comment]
Hello @sdktr 

After upgrading our nodes from E2s_v4 to E2as_v4 DNS latency dropped to a reasonable amount.

After enabling CoreDNS logs we analyzed all queries, and most of the above 1 ms are about external hosts (Not within cluster), which might explain higher average latency.

On average the A and AAAA response times are slower and above 1ms which triggered Prometheus alert.

We are going to increase the threshold to 10 ms which is more reasonable. Closing the issue.
